### ISSUE TITLE
[Due for payment 2025-02-18] [HOLD for payment 2025-02-04] Tweaks to the `Create Expense` Flow 

### ISSUE THREAD

**Issue by trjExpensify** [Expensify/App#52981]
Coming from [here](https://expensify.slack.com/archives/C07HPDRELLD/p1731690049777719?thread_ts=1731604883.924969&cid=C07HPDRELLD). 

**Background:** We're experimenting with an A/B test of a combined "Create expense" flow to see if we improve the decision juncture between "Submit expense" and "Track expense" out the gate. We've seen in Fullstory that submitters are still getting wrong-footed on this page choosing the "Just track it (don't submit it!)" option row. All the while, the workspace settings actually determine whether an expense created is submitted on creation, or if it’s added to a draft report to be submitted later, either manually or automatically with scheduled submit.

**Problem:** When submitters choose the "Just track it (don't submit it!)" on this page, they end up creating non-reimbursable expenses in the selfDM, which leads to confusion and having no recourse to getting paid back. 

**Solution:** Let's make a couple of tweaks to the `Create expense` flow, as follows:

**Participant selector screen**
- Title it "Choose recipient"
- Remove the "Just track it (don't submit it)" option row 
- Add a section for "Workspaces" at the top (if you're a member of a workspace). Your primary workspace should be in the top position. That should be followed by a section for "Personal", which has your selfDM. Then followed by the existing "Recents" (if you have any that aren't already above in workspaces/selfDM) and "Contacts" sections. 

**Confirmation page screen**
- Title it "Confirm details" 
- Update the confirmation button to read: "Create $amount expense"

![image](https://github.com/user-attachments/assets/8bf2879f-d43c-4578-b69b-2431b19593b0)

CC: @danielrvidal @JmillsExpensify @Expensify/design 

### New addition from @dubielzyk-expensify to add as a follow-up
In the participant selector, switch out the second line of each workspace row (which currently reads `Workspace`) to be the displayName of the approver ("`submitsTo`") instead. Use the primaryLogin as a fallback if no displayName is set. 

<img width="222" alt="image" src="https://github.com/user-attachments/assets/f3d6c645-1444-434b-9b04-51dc93a8f1de" />

<details><summary>Issue Owner</summary>Current Issue Owner: @trjExpensify</details>

**Comment by mkzie2** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Tweaks to the Create Expense Flow

### What is the root cause of that problem?
New feature
### What changes do you think we should make in order to solve the problem?
- The test branch here: https://github.com/mkzie2/App/tree/mkzie2-issue/52981-temp

1. Create a new translation for `Choose recipient` and `Confirm details`

2. Participant selector screen

> Title it "Choose recipient"
- Change `headerTitle` to `Choose recipient` or only change to it on the default case and the if `action === CONST.IOU.ACTION.SUBMIT`

https://github.com/Expensify/App/blob/3ebe8520d74c57a2fe9548b7d4307206e2b7c673/src/pages/iou/request/step/IOURequestStepParticipants.tsx#L47

> Remove the "Just track it (don't submit it)" option row
- Remove `shouldDisplayTrackExpenseButton` and `onTrackExpensePress` props in `MoneyRequestParticipantsSelector` and all variables and functions related with these props

https://github.com/Expensify/App/blob/3ebe8520d74c57a2fe9548b7d4307206e2b7c673/src/pages/iou/request/step/IOURequestStepParticipants.tsx#L191-L194

> Add a section for "Workspaces" at the top (if you're a member of a workspace). Your primary workspace should be in the top position. That should be followed by a section for "Personal", which has your selfDM. Then followed by the existing "Recents" (if you have any that aren't already above in workspaces/selfDM) and "Contacts" sections.

  - Introduce two new params in `getOptions` and `getFilteredOptions` function `shouldSeparateSelfDMChat`, `shouldSeparateWorkspaceChat`. The default value is `false`.  In `getOptions`, if it's true we will return recent workspace chats/selfDM chats in a separate variable
  
    - Pass two params above as true [here](https://github.com/Expensify/App/blob/3ebe8520d74c57a2fe9548b7d4307206e2b7c673/src/pages/iou/request/MoneyRequestParticipantsSelector.tsx#L108)
  - Update `filterOptions` to implement the splice and search logic for `recentWorkspaceChats` like `recentReports`

https://github.com/Expensify/App/blob/3ebe8520d74c57a2fe9548b7d4307206e2b7c673/src/libs/OptionsListUtils.ts#L1743
  - Merge `recentWorkspaceChats` and `recentReports` options when we call `formatSectionsFromSearchTerm` here or we can introduce a new param `recentWorkspaceChats` in this function
  
  https://github.com/Expensify/App/blob/3ebe8520d74c57a2fe9548b7d4307206e2b7c673/src/pages/iou/request/MoneyRequestParticipantsSelector.tsx#L164-L169

   - Pick `isSelfDM` field here, and if `isSelfDM` is [true](https://github.com/Expensify/App/blob/3ebe8520d74c57a2fe9548b7d4307206e2b7c673/src/pages/iou/request/MoneyRequestParticipantsSelector.tsx#L231) we will not call `onFinish` function.
   - In this function we will call `trackExpense` function if the first option is selfDM
   
   ```
const firstParticipant = val.at(0);

            if (firstParticipant?.isSelfDM) {
                trackExpense();
                return;
            }
   ```

https://github.com/Expensify/App/blob/3ebe8520d74c57a2fe9548b7d4307206e2b7c673/src/pages/iou/request/step/IOURequestStepParticipants.tsx#L87

3. Confirmation page screen

> Title it "Confirm details"
- Change the `headerTitle` to `Confirm details` or only do it in the case Track/submit expense

https://github.com/Expensify/App/blob/3ebe8520d74c57a2fe9548b7d4307206e2b7c673/src/pages/iou/request/step/IOURequestStepConfirmation.tsx#L111

> Update the confirmation button to read: "Create $amount expense"
-  Create a new translation for `Create ${amount} expense` and use this translation [here](https://github.com/Expensify/App/blob/3ebe8520d74c57a2fe9548b7d4307206e2b7c673/src/components/MoneyRequestConfirmationList.tsx#L412) and [here](https://github.com/Expensify/App/blob/3ebe8520d74c57a2fe9548b7d4307206e2b7c673/src/components/MoneyRequestConfirmationList.tsx#L421-L424
)

Optional: For changing the title and the submit button, we can only do this if `canUseCombinedTrackSubmit` is `true`
### What alternative solutions did you explore? (Optional)

### Result


https://github.com/user-attachments/assets/da91342b-5671-4342-bf1d-9ae107cb3075

**Comment by grgia** [comment]
I reached out to Fábio about taking this 💬

**Comment by trjExpensify** [comment]
Popped on planning for a sec, discussing [here](
https://expensify.slack.com/archives/C07HPDRELLD/p1732537181702459?thread_ts=1731604883.924969&cid=C07HPDRELLD).

**Comment by trjExpensify** [comment]
No change, thread likely picked up next week at this point.

**Comment by trjExpensify** [comment]
Anu is back today, bumped [the thread](https://expensify.slack.com/archives/C07HPDRELLD/p1732537181702459?thread_ts=1731604883.924969&cid=C07HPDRELLD).

**Comment by JmillsExpensify** [comment]
Nice, I'd like to get this one moving again. I'm still concerned that the original `Create Expense` test isn't testing what we need to test (e.g. whether they would like to submit now or later, which is different than whether they don't want to submit to anyone. ever).

**Comment by grgia** [comment]
@JmillsExpensify @trjExpensify could you ping me when this is ready to be picked up again by an engineer

**Comment by trjExpensify** [comment]
Yep, will do.

**Comment by JKobrynski** [comment]
Hi, I'm Julian from Callstack - expert agency - and I would like to work on this issue.

**Comment by trjExpensify** [comment]
How you getting on here, @JKobrynski? Everything clear?

**Comment by JKobrynski** [comment]
@trjExpensify working on it right now, I will be posting a proposal/PR today!

**Comment by trjExpensify** [comment]
Excellent, thanks!

**Comment by JKobrynski** [comment]
@trjExpensify unfortunately I wasn't able to finish everything today, sorry for that 😞 it will be my top priority for tomorrow to get this done, I will keep you posted. I already created a [draft PR](https://github.com/Expensify/App/pull/53760/files) but I still have to add these new sections to the participants list.

**Comment by trjExpensify** [comment]
Awesome, appreciate the update!

**Comment by JKobrynski** [comment]
@trjExpensify the [PR](https://github.com/Expensify/App/pull/53760) is ready for review!

**Comment by flaviadefaria** [comment]
I'm also adding this to #migrate and making it a daily as we believe this will prevent incorrect submissions from migrated customers which could be a problem for submissions in the holiday period. More context [here](https://expensify.slack.com/archives/C07NMDKEFMH/p1734033662626919?thread_ts=1734029325.871339&cid=C07NMDKEFMH).

**Comment by JKobrynski** [comment]
I haven't had time to update the PR today, I will be working on it tomorrow.

**Comment by JKobrynski** [comment]
PR is still in review, I asked for some clarification to one of the review comments which I tried to implement but couldn't fully understand what needs to be done.

**Comment by JKobrynski** [comment]
Today I made updates to the PR, they were mostly about simplifying workspace filtering. There were also many conflicts that directly affected the new code, I fixed them too. I'm waiting for another round of review.

**Comment by JKobrynski** [comment]
Update: I resolved conflicts that appeared today and addressed one review comment. The PR is awaiting review.

**Comment by JKobrynski** [comment]
Update: I implemented requested changes, resolved conflicts and updated the code not to use `ReportUtils.getReport`.

**Comment by JKobrynski** [comment]
Update:

- Fixed merge conflicts
- Removed string id fallbacks based on @mountiny 's comments (updated methods if necessary)

**Comment by JKobrynski** [comment]
Update:

@VickyStash  reached out to me today to let me know that all files listed in .eslint.changed.js override list will be migrated later to fit the latest fallback id rule, so I reverted all related changes that I've made to OptionsListUtis.ts. The PR is waiting for final review.

**Comment by JKobrynski** [comment]
Update:

Today I made some minor updates to the PR based on Vicky's comments. The PR is still waiting for final review. It's been approved by Vicky.

**Comment by JKobrynski** [comment]
Update:

Earlier today, a final test build was triggered to conduct the final test.

**Comment by trjExpensify** [comment]
Awesome, so close! Once we've merged that PR, can you look at this follow-up from the OP next? 

> ### New addition from @dubielzyk-expensify to add as a follow-up
> In the participant selector, switch out the second line of each workspace row (which currently reads `Workspace`) to be the displayName of the approver ("`submitsTo`") instead. Use the primaryLogin as a fallback if no displayName is set. 
> 
> <img width="222" alt="image" src="https://github.com/user-attachments/assets/f3d6c645-1444-434b-9b04-51dc93a8f1de" />

**Comment by trjExpensify** [comment]
Let me take off the payment holds on this issue. [PR](https://github.com/Expensify/App/pull/55016) in review here.

**Comment by JKobrynski** [comment]
Update:

I fixed merge conflicts, lint issues and applied changes requested by reviewers. The PR is once again waiting for review.

**Comment by JKobrynski** [comment]
Update:

I applied a small change requested by @trjExpensify. Unfortunately new conflicts appeared right after that. They couldn't be fixed quickly as they directly influence the changes that I introduced in `OptionsListUtils.ts`. I will continue my work on it tomorrow.

**Comment by trjExpensify** [comment]
Mergeeeed!

**Comment by trjExpensify** [comment]
@JKobrynski jus' making sure this is still on your radar: https://github.com/Expensify/App/issues/52981#issuecomment-2569050851

**Comment by JKobrynski** [comment]
@trjExpensify should we create an issue for that? 

CC: @grgia

**Comment by trjExpensify** [comment]
Not passionate, personally. It's in the OP of this one currently. 👍

**Comment by grgia** [comment]
@JKobrynski would it help to create an issue?

**Comment by JKobrynski** [comment]
@grgia no need! I'm on it

**Comment by JKobrynski** [comment]
@mananjadhav @grgia 

The [PR](https://github.com/Expensify/App/pull/55863) related to https://github.com/Expensify/App/issues/52981#issuecomment-2569050851 is ready for review!

**Comment by trjExpensify** [comment]
Wahoo! 🙏

**Comment by JKobrynski** [comment]
Update:

The PR is in review

**Comment by JKobrynski** [comment]
Update:

I haven't had time to work on this issue today. I'm going to work on it tomorrow.

**Comment by trjExpensify** [comment]
Great stuff, I'm looking forward to getting that one over the line. Thanks!

**Comment by mananjadhav** [comment]
I helped with 2 PR reviews here. So there'll be a payout for me.

**Comment by JKobrynski** [comment]
Update:

I studied Manan's comments on the PR, tried reproducing the issue he had but failed. I then examined the code and debugged it to try to figure out the reason behind this unwanted behavior. I suggested some next steps we can take.

**Comment by JKobrynski** [comment]
Update:

I fixed the conflicts and ran the performance tests requested by reviewer and posted the results in the PR.

**Comment by trjExpensify** [comment]
>I helped with 2 PR reviews here. So there'll be a payout for me.

Yeah agreed, figured we'll settle up once we're through this last PR.

**Comment by JKobrynski** [comment]
Update:

I addressed reviewer's comment and resolved merge conflicts.

**Comment by mananjadhav** [comment]
Just adding for tracking: There are 3 payments pending on this one for the PR reviews done. 2 PRs linked in this issue and 1 from [here](https://github.com/Expensify/App/issues/55621)

**Comment by mananjadhav** [comment]
There's this one edge case https://github.com/Expensify/App/issues/56507 left out related to the subtitle change. This can be treated as regression to the subtitle PR https://github.com/Expensify/App/pull/55863

**Comment by trjExpensify** [comment]
Okay cool, so that's $625 total due then, right?

**Comment by trjExpensify** [comment]
Payment summary as follows: 

- $625 to @mananjadhav for the PR reviews 

Go ahead and request! I've created a regression test request for the submitsTo subtitle inclusion. Thanks!

**Comment by mananjadhav** [comment]
Thanks for the summary. I was going to request if we could pay this in full. 

The regression case was a very edge case and thought surfaced with the PR, but it's because the Options list doesn't refresh. Hence, I didn't have a way to compare it with the production behavior too.

**Comment by trjExpensify** [comment]
Oh right, then I misinterpreted you here: 

>There's this one edge case https://github.com/Expensify/App/issues/56507 left out related to the subtitle change. This can be treated as regression to the subtitle PR https://github.com/Expensify/App/pull/55863.

@grgia what do you think about the below wrt to the regression? 

>The regression case was a very edge case and thought surfaced with the PR, but it's because the Options list doesn't refresh. Hence, I didn't have a way to compare it with the production behavior too.

**Comment by mananjadhav** [comment]
Yeah I didn't think about it while writing the comment. I realized when I was reviewing the PR.

**Comment by mananjadhav** [comment]
@grgia Can you please check the previous conversations?

**Comment by grgia** [comment]
@trjExpensify I agree with @mananjadhav

**Comment by trjExpensify** [comment]
https://github.com/Expensify/App/issues/52981#issuecomment-2669731488 - cool amended payment summary to $750.

**Comment by garrettmknight** [comment]
$750 approved for @mananjadhav
